packages/cozy-pouch-link/src/PouchManager.js

```diff
@@ -137,6 +147,16 @@ export default class PouchManager {
         () => this.startReplicationLoop(),
         false
       )
+      document.removeEventListener(
+        'offline',
+        () => this.stopReplicationLoop(),
+        false
+      )
+      document.removeEventListener(
+        'online',
+        () => this.startReplicationLoop(),
+        false
+      )
```

Comment:
The event listeners are not removed here, since you must pass to removeEventListener the original function that was bound.

Here a closure is created when adding the event listener, and a reference to this closure is not saved anywhere, preventing you to properly remove it. I think the creation of the closure is to have `this` correctly bound. An alternative to this is to bound the method in the constructor and then use the event handlers API with this method (directly without wrapping in a closure).

```
constructor () {
  this.startReplicationLoop = this.startReplicationLoop.bind(this)
  this.stopReplicationLoop = this.stopReplicationLoop.bind(this)
}

addEventListeners() {
  document.addEventListener('online', this.startReplicationLoop)
}

removeEventListeners() {
  document.removeEventListener('online', this.startReplicationLoop)
}
```